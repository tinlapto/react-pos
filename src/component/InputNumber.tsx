import { Input } from "antd";
import { MinusOutlined, PlusOutlined } from "@ant-design/icons";

const InputNumber = ({
  value,
  onChange,
  min,
  max,
}: {
  value: number;
  onChange: (number: number) => void;
  min?: number;
  max?: number;
}) => {
  const handleOnChange = (value: any) => {
    if (!Number.isInteger(value)) return;
    if (max !== undefined && value > max) return;
    if (min !== undefined && value < min) return;
    onChange(value);
  };
  return (
    <Input
      style={{ textAlign: "right" }}
      value={value}
      onChange={(e) => handleOnChange(Number(e.target.value))}
      addonBefore={<PlusOutlined onClick={() => handleOnChange(value + 1)} />}
      addonAfter={<MinusOutlined onClick={() => handleOnChange(value - 1)} />}
    />
  );
};

export default InputNumber;
