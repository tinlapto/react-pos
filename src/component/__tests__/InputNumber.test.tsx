import {
  render,
  screen,
  fireEvent,
  queryByAttribute,
} from "@testing-library/react";
import InputNumber from "../InputNumber";

const getByClass = queryByAttribute.bind(null, "class");

describe("InputNumber", () => {
  test("render", () => {
    const onChange = jest.fn();
    const { asFragment } = render(
      <InputNumber value={1} onChange={onChange} />
    );
    expect(asFragment()).toMatchSnapshot();
    fireEvent.click(screen.getByLabelText("plus"));
    expect(onChange).toBeCalledWith(2);
    fireEvent.click(screen.getByLabelText("minus"));
    expect(onChange).toBeCalledWith(0);
  });

  test("render max", () => {
    const onChange = jest.fn();
    render(<InputNumber value={1} onChange={onChange} max={1} />);
    fireEvent.click(screen.getByLabelText("plus"));
    expect(onChange).not.toBeCalled();
  });

  test("render min", () => {
    const onChange = jest.fn();
    render(<InputNumber value={1} onChange={onChange} min={1} />);
    fireEvent.click(screen.getByLabelText("minus"));
    expect(onChange).not.toBeCalled();
  });

  test("input number", () => {
    const onChange = jest.fn();
    const { container } = render(
      <InputNumber value={1} onChange={onChange} min={1} />
    );
    fireEvent.change(getByClass(container, "ant-input") as Element, {
      target: { value: 10 },
    });
    expect(onChange).toBeCalledWith(10);
  });

  test("input text", () => {
    const onChange = jest.fn();
    const { container } = render(
      <InputNumber value={1} onChange={onChange} min={1} />
    );
    fireEvent.change(getByClass(container, "ant-input") as Element, {
      target: { value: "abc" },
    });
    expect(onChange).not.toBeCalled();
  });
});
