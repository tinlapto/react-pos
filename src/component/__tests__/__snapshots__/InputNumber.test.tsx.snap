// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`InputNumber render 1`] = `
<DocumentFragment>
  <span
    class="ant-input-group-wrapper"
    style="text-align: right;"
  >
    <span
      class="ant-input-wrapper ant-input-group"
    >
      <span
        class="ant-input-group-addon"
      >
        <span
          aria-label="plus"
          class="anticon anticon-plus"
          role="img"
          tabindex="-1"
        >
          <svg
            aria-hidden="true"
            data-icon="plus"
            fill="currentColor"
            focusable="false"
            height="1em"
            viewBox="64 64 896 896"
            width="1em"
          >
            <defs>
              <style />
            </defs>
            <path
              d="M482 152h60q8 0 8 8v704q0 8-8 8h-60q-8 0-8-8V160q0-8 8-8z"
            />
            <path
              d="M176 474h672q8 0 8 8v60q0 8-8 8H176q-8 0-8-8v-60q0-8 8-8z"
            />
          </svg>
        </span>
      </span>
      <input
        class="ant-input"
        type="text"
        value="1"
      />
      <span
        class="ant-input-group-addon"
      >
        <span
          aria-label="minus"
          class="anticon anticon-minus"
          role="img"
          tabindex="-1"
        >
          <svg
            aria-hidden="true"
            data-icon="minus"
            fill="currentColor"
            focusable="false"
            height="1em"
            viewBox="64 64 896 896"
            width="1em"
          >
            <path
              d="M872 474H152c-4.4 0-8 3.6-8 8v60c0 4.4 3.6 8 8 8h720c4.4 0 8-3.6 8-8v-60c0-4.4-3.6-8-8-8z"
            />
          </svg>
        </span>
      </span>
    </span>
  </span>
</DocumentFragment>
`;
